packages/client/src/components/JoinPath/JoinPath.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Modal } from '@codement/ui/components/Modal/Modal';
+import { Icon, PathIcon, Button } from '@codement/ui';
+import gql from 'graphql-tag';
+import { useQuery } from '@apollo/react-hooks';
+import { Path } from '@codement/api';
+import icons from '@codement/ui/components/PathIcon/path-icons/icons';
+
+// TODO: Handle selection of paths
+// TODO: Disable begin button until path selected
+
+const allPathsQuery = gql`query {
+  paths { id name icon }
+}`;
+
+interface JoinPath {
+  setShow: any;
+}
+
+// Click add path button in progress widget to open modal
+
+// TEAM NOTE:
+//  - Modal closes when clicked anywhere.  Changed to only close on (x).
+//  - Refractor CSS to its own individual file.
+//  - Added the fetch paths feature.(create the paths in graphql playground to enable functionality)
+//  - Overtiding icon type in PathIcon component using 'as'
+//  - I think we should probably make separate css files for the each component rather than 1 in UI
+
+// CSS can be found @     ui/css/modal-join-path.css
+// Figma Design:          https://www.figma.com/file/eyk5tQgLhIpyiORYfWxeUh/Learning-App?node-id=197%3A0
+
+export const JoinPath: React.FC<JoinPath> = ({ setShow }) => {
+
+  const { data } = useQuery<{paths: Path[]}>(allPathsQuery);
+
+  return (
+    <Modal setShow={setShow}>
+      <div className="join-path-card">
+        <Icon
+          icon="x"
+          size="small"
+          className="btn-close-icon"
+          onClick={() => setShow()}
+        />
+
+        <h1 className="mb-3 font-semibold">Join a Path</h1>
+
+        <p className="text-sm leading-4 text-grey-800 mb-12" style={{ letterSpacing: '0.75px' }}>
```

Comment:
Something small like this, I don't think I mind to have within the "HTML", but **I think** @tristanMatthias prefers the CSS to be in a separate file.
* Not trying to put words in his mouth, it's just something he mentioned before because he wants to keep the "HTML" as clean and simple as possible. But maybe here he won't mind...let's wait for his answer...
=)

Reply:
Thanks Suzy :)